File: problems/b/bela/bela.cpp
/*
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
*/

#include <iostream>

using namespace std;

int main() {
    int n, hands, points;
    char b, card_n, card_b;

    cin >> n;
    cin >> b;

    hands = n * 4;
    points = 0;

    for (int i=0; i<hands; i++) {
        cin >> card_n;
        cin >> card_b;
        
        switch (card_n) {
            case 'A':
                points += 11;
                break;
            case 'K':
                points += 4;
                break;
            case 'Q':
                points += 3;
                break;
            case 'J':
                if (card_b == b) {
                    points += 20;
                }
                else {
                    points += 2;
                }
                break;
            case 'T':
                points += 10;
                break;
            case '9':
                if (card_b == b) {
                    points += 14;
                }
                else {
                    points += 0;
                }
                break;
            case '8':
                points += 0;
                break;
            case '7':
                points += 0;
                break;
        }
    }

    cout << points << endl;
}

File: problems/b/bela/bela.py
#! /usr/bin/python3
"""
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
"""

def main():
    dominant = {'A':11, 'K':4, 'Q':3, 'J':20, 'T':10, '9':14, '8':0, '7':0}
    not_dominant = {'A':11, 'K':4, 'Q':3, 'J':2, 'T':10, '9':0, '8':0, '7':0}

    nb = input().split()
    n = int(nb[0]) * 4
    b = nb[1]
    points = 0

    for _ in range(n):
        card = input().replace('\n','')
        if card[1] == b:
            points += dominant[card[0]]
        else:
            points += not_dominant[card[0]]

    print(points)


if __name__ == '__main__':
    main()